src/palace.pyx

```diff
@@ -179,9 +178,8 @@ cdef class Device:
     """
     cdef alure.Device impl
 
-    def __init__(self, name: Optional[str] = '',
+    def __init__(self, name: str = '',
                  fail_safe: bool = False) -> None:
```

Comment:
These 2 lines are short enough to be joined into one.

Reply:
this is 72 as I calculated, is it OK for generating docs?

Reply:
72 is the limit for (multiline) docstrings, for codes limit to 79.

Reply:
Oof, it is 73, but I merged it either

Reply:
It's code so it's OK.  BTW read the comment above and keep trying till the tests pass.

Reply:
But how could I fix those, simply delete that?

Reply:
`Source(None)` should be changed into `Source.__new__(Source)` as you did for some of them.

Reply:
I did the changes but when I compile, it fails